### DISCUSSION TITLE
the trait IntoResponse is not implemented for CustomError in workspace

### DISCUSSION THREAD

**Discussion by zkrzyzanowski** [tokio-rs/axum#2556] [Q&A] [answered]
### Summary

minimal repro here: https://github.com/zkrzyzanowski/axum-trait-error

I've implemented a `CustomError` type as part of a `lib` named `error`. In my binary, `serve`, I create a minimal handler [here](https://github.com/zkrzyzanowski/axum-trait-error/blob/33dcd7eef679b2b870e36b42f3ecddd193e4c73e/serve/src/main.rs#L34) that returns a `Result<String, CustomError>`. All of these are part of a workspace.

However, this gives a compiler error on the `Result` type annotation
```
 the trait `IntoResponse` is not implemented for `CustomError`
```

However, if you implement the `CustomError` type in the main binary by uncommenting the code [here](https://github.com/zkrzyzanowski/axum-trait-error/blob/33dcd7eef679b2b870e36b42f3ecddd193e4c73e/serve/src/main.rs#L38) in `main.rs` and comment out the import of the `error` lib, the code compiles fine.

I was previously on `0.6` of axum and did not have this issue. 

Any help would be appreciated!

### axum version

0.7.4

**Comment by jplatte** [comment, marked as answer]
Your error create still depends on axum 0.6. You need to depend on the same version of axum (or really, on two versions that use the same axum-core version) in both crates.

**Reply by zkrzyzanowski** [reply]
thank you!! sorry for the obvious mistake
